iOS: Support GDPR (APSAnalytics 2.1.0), add missing native header

hansemannn opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26004

Adding unit test once CR approved.

hansemannn committed:
Support GDPR (APSAnalytics 2.1.0)

build commented:
<!--
  1 failure:  :microscope: Ther...
  0 warning: 
  1 messages
  
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/hansemannn/titanium_mobile/blob/TIMOB-26004/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10051/1/artifact/dist/mobilesdk-7.3.0.v20180516042351-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

hansemannn committed:
Merge branch 'master' into TIMOB-26004

hansemannn committed:
Merge branch 'master' into TIMOB-26004

garymathews reviewed (dismissed):
I think we need to agree on the API specification for Ti.Analytics optOut because this is varying per implementation:
```
Ti.Analytics.optedOut // instead of Ti.Analytics.isOptedOut
Ti.Analytics.setOptedOut(boolean)
Ti.Analytics.getOptedOut()
```

For consistency like our other Titanium APIs

For example, the Windows implementation uses `isOptOut`:
https://github.com/appcelerator/titanium_mobile_windows/pull/1238

And Android uses `optedOut`:
https://github.com/appcelerator/titanium_mobile/pull/10058

hansemannn committed:
Change getter, update docs

hansemannn commented:
@garymathews @ypbnv @infosia I have updated the iOS-PR to add the property `optedOut` with it's getter `getOptedOut()` and setter `setOptedOut(true|false)`. I have also added the docs for iOS and Android - I think the "windows" platform tag needs to be added via the titanium_mobile_windows repo?

infosia commented:
Updated Windows PR: https://github.com/appcelerator/titanium_mobile_windows/pull/1238 Updated API doc for Windows too.

hansemannn committed:
Update binary based on review in aps_sdk repo

hansemannn committed:
Extend GDPR description, use final binary

hansemannn committed:
Do not throw.

hansemannn committed:
Merge branch 'master' into TIMOB-26004

hansemannn commented:
@garymathews The PR was updated, can you re-review? I have also added @vijaysingh-axway for another CR.

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-26004

vijaysingh-axway commented on iphone/Classes/AnalyticsModule.m:
```diff
@@ -105,6 +105,17 @@ - (void)filterEvents:(id)args
+
+- (void)optedOut
+{
+  @([[APSAnalytics sharedInstance] isOptedOut]);
```
1. Isn't it good if we make this method's return type to (NSNumber *) for more readability and return it.
2. Following test case crashing, if I close the app and run again. Can this not be a test case ?
`var win = Ti.UI.createWindow({
    backgroundColor: '#fff'
});

var btn = Ti.UI.createButton({
    title: 'Trigger'
});

btn.addEventListener('click', function() {
	Ti.Analytics.setOptedOut(true);
    });

win.add(btn);
win.open();
`

vijaysingh-axway reviewed (changes requested):
@hansemannn Can you address the 2 points which I have mentioned. Thanks!

hansemannn committed:
Fix stupid return typo, do not throw on re-init

hansemannn committed:
Use updated APSAnalytics binary

hansemannn committed:
Add unit-test

hansemannn commented on tests/ti.analytics.addontest.js:
```diff
@@ -0,0 +1,32 @@
+var should = require('./utilities/assertions');
+
+describe('Titanium.Analytics', function () {
+	it.androidMissing('.optedOut', function () {
```
@garymathews @ypbnv Once Android is merged, we can update this one to work or all platforms.

vijaysingh-axway reviewed (approved):
CR passed.

build commented:
<!--
  1 failure:  Tests have failed...
  0 warning: 
  1 messages
  1 markdown notices
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Tests have failed, see below for more information.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10051/11/artifact/dist/mobilesdk-7.3.0.v20180524005701-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>

### Tests: 

| Classname | Name | Time | Error |
| --- | --- | --- | --- |
| android.Titanium.UI.TableView | Add and remove headerView/footerView  | 3.421 | Error: timeout of 2000ms exceeded |

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

hansemannn committed:
Use final APSAnalytics SDK

hansemannn committed:
Update ti.analytics.addontest.js
